pkg/ingester/transfer.go

```diff
@@ -214,6 +214,13 @@ func (i *Ingester) TransferTSDB(stream client.Ingester_TransferTSDBServer) error
 	fromIngesterID := ""
 
 	xfer := func() error {
+
+		dir, err := ioutil.TempDir("", "tsdb_xfer")
+		if err != nil {
+			return err
+		}
+		defer os.Remove(dir)
```

Comment:
Two things:
1. This expects that `dir` is empty. I think you should use `RemoveAll()` in this case.
2. Do we really want to remove it? On success shouldn't exist anymore because it has been moved, while on failure may be handy to keep them for debugging? _It's an open question, not sure_

Reply:
Yea, removall would be correct. I'm not sure what would be gained debug wise from keeping it though.

Reply:
Ok, `RemoveAll()` would be then 😉

Reply:
Changed it to removeall